raiden/network/transport/matrix/transport.py

```diff
@@ -550,24 +553,35 @@ def send_global(self, room: str, message: Message) -> None:
 
         assert self._global_rooms.get(room_name), f'Unknown global room: {room_name!r}'
 
-        def _send_global():
-            text = JSONSerializer.serialize(message)
-            room = self._global_rooms[room_name]
-            self.log.debug(
-                'Send global',
-                room_name=room_name,
-                room=room,
-                data=text.replace('\n', '\\n'),
-            )
-            room.send_text(text)
-
-        greenlet = self._spawn(_send_global)
-        greenlet.name = (
-            f'MatrixTransport.send_global '
-            f'node:{pex(self._raiden_service.address)} '
-            f'room:{room} '
-            f'user_id:{self._user_id}'
-        )
+        self._global_send_queue.put((room_name, message))
+
+    def _global_send_worker(self):
```

Comment:
I don't think we need a new greenlet + new worker which should live exactly while transport runnable is running, as `MatrixTransport._run` already sits there doing nothing but waiting on the `_stop_event` to be set. You can use it `_run` to loop on the messages instead.